raster/r.sim/simlib/simlib.h

```diff
@@ -4,38 +4,55 @@
 /*! \file simlib.h
  * \brief This is the interface for the simlib (SIMWE) library.
  */
+#include <stdbool.h>
 
 #define NUM_THREADS "1"
 #if defined(_OPENMP)
 #include <omp.h>
 #endif
 
-struct WaterParams {
-    double xmin, ymin, xmax, ymax;
-    double mayy, miyy, maxx, mixx;
+typedef struct {
     int mx, my;
-    int mx2, my2;
-
-    double bxmi, bymi, bxma, byma, bresx, bresy;
-    int maxwab;
-    double step, conv;
-
-    double frac;
+    double xmin, xmax, ymin, ymax;
+    double miyy, mixx;
+    double step, stepx, stepy;
+    double conv;
+    double xp0, yp0;
+} Geometry;
 
+typedef struct {
+    double halpha;
     double hbeta;
-    double hhmax, sisum, vmean;
-    double infsum, infmean;
-    int maxw, maxwa, nwalk;
-    double rwalk, xrand, yrand;
-    double stepx, stepy, xp0, yp0;
-    double chmean, si0, deltap, deldif, cch, hhc, halpha;
-    double eps;
-    int nstack;
-    int iterout, mx2o, my2o;
-    int miter, nwalka;
-    double timec;
-    int ts, timesec;
+    double hhmax;
+    double frac; /* Water diffusion constant */
```

Comment:
You may consider using `//` style trailing comments here in this file, "cleaner" and saves 3 characters!